google_cloud_automlops/orchestration/kfp/KFPComponent.py

```diff
@@ -0,0 +1,210 @@
+
+        # Set packages to install and component spec attributes
+        self.packages_to_install_command = self._get_packages_to_install_command()
+        self.component_spec = self._create_component_spec()
+
+    def build(self):
+        """Constructs files for running and managing Kubeflow pipelines.
+        """
+        defaults = read_yaml_file(GENERATED_DEFAULTS_FILE)
+        self.artifact_repo_location = defaults['gcp']['artifact_repo_location']
+        self.artifact_repo_name = defaults['gcp']['artifact_repo_name']
+        self.project_id = defaults['gcp']['project_id']
+        self.naming_prefix = defaults['gcp']['naming_prefix']
+
+        # Set and create directory for components if it does not already exist
+        component_dir = BASE_DIR + 'components/' + self.component_spec['name']
+
+        # Build necessary folders
+        # TODO: make this only happen for the first component? or pull into automlops.py
+        make_dirs([
+            component_dir,
+            BASE_DIR + 'components/component_base/src/'])
+
+        # TODO: can this be removed?
+        kfp_spec_bool = self.component_spec['implementation']['container']['image'] != PLACEHOLDER_IMAGE
+
+        # Read in component specs
+        custom_code_contents = self.component_spec['implementation']['container']['command'][-1]
+        compspec_image = (
+                f'''{self.artifact_repo_location}-docker.pkg.dev/'''
+                f'''{self.project_id}/'''
+                f'''{self.artifact_repo_name}/'''
+                f'''{self.naming_prefix}/'''
+                f'''components/component_base:latest''')
+
+        # If using kfp, remove spaces in name and convert to lowercase
+        if kfp_spec_bool:
+            self.component_spec['name'] = self.component_spec['name'].replace(' ', '_').lower()
+
+        # Write task script to component base
+        write_file(
+            filepath=BASE_DIR + 'components/component_base/src/' + self.component_spec['name'] + '.py',
+            text=render_jinja(
+                template_path=import_files(KFP_TEMPLATES_PATH + '.components.component_base.src') / 'task.py.j2',
+                generated_license=GENERATED_LICENSE,
+                kfp_spec_bool=kfp_spec_bool,
+                custom_code_contents=custom_code_contents),
+            mode='w')
+
+        # Update component_spec to include correct image and startup command
+        self.component_spec['implementation']['container']['image'] = compspec_image
+        self.component_spec['implementation']['container']['command'] = [
+            'python3',
+            f'''/pipelines/component/src/{self.component_spec['name']+'.py'}''']
+
+        # Write license and component spec to the appropriate component.yaml file
+        comp_yaml_path = component_dir + '/component.yaml'
+        write_file(
+            filepath=comp_yaml_path,
+            text=GENERATED_LICENSE,
+            mode='w')
+        write_yaml_file(
```

Comment:
What's the difference between write_file and write_yaml_file? Could there be a type added to the argument of write_file to support yaml? 
Such as: 
```     
write_file(
            filepath=comp_yaml_path,
            text=GENERATED_LICENSE,
            mode='w',
            type='YAML')
```